injected/src/features/web-compat.js

```diff
@@ -867,6 +867,17 @@ export class WebCompat extends ContentFeature {
         return deviceInfo;
     }
 
+    /**
+     * Helper to wrap a promise with timeout
+     * @param {Promise} promise - Promise to wrap
+     * @param {number} timeoutMs - Timeout in milliseconds
+     * @returns {Promise} Promise that rejects on timeout
+     */
+    withTimeout(promise, timeoutMs) {
```

Comment:
We may wish to move this to the messaging layer for any await'd requests at least for Windows. For now this is the most prevalent message we're triggering in web context Windows.